samcli/local/layers/layer_downloader.py

```diff
@@ -106,7 +107,8 @@ def download(self, layer: LayerVersion, force=False) -> LayerVersion:
             LOG.info("%s is already cached. Skipping download", layer.arn)
             return layer
 
-        layer_zip_path = layer.codeuri + ".zip"
+        current_thread_id = str(threading.get_ident())
+        layer_zip_path = f"{layer.codeuri}_{current_thread_id}.zip"
```

Comment:
If we are looking for getting a unique value, why we are not using `uuid`?

Reply:
Oh I was thinking about one file per thread, but yeah using `uuid` is also good for this case.